### ISSUE TITLE
Waiting list: Make ordering of more than one product possible

### ISSUE THREAD

**Issue by Unicorn-rzl** [pretix/pretix#974]
The current implementation of the waiting list requires visitors to reserve each ticket individually. This has a potential issue:
If a group of people (n>1) wants to attend they don't know whether all will get tickets. For example: If you imagine a parent with kid where either both or none can attend, they would be forced to wait for both reservations to be available. Depending on the voucher validity, the first voucher could already be invalid.
One solution would be to allow more than one ticket in one reservation. However, this could lead to some tickets being not send out (Example: Next place on waiting list wants 5 tickets, however only 3 are available. This detail could be circumvented by manually sending out vouchers in the end.

**Comment by raphaelm** [comment]
The use case is clear and often requested.

> However, this could lead to some tickets being not send out (Example: Next place on waiting list wants 5 tickets, however only 3 are available. This detail could be circumvented by manually sending out vouchers in the end.

Manually sending out is not good enough, and a human will face the same challenge as an algorithm deciding this: It is unclear, if more seats will become free, so it might be better to give the tickets to someone lower on the list to avoid the seats from staying unused and therefore missing revenue. As long as there isn't a great idea how to solve this, I don't really see this change happening :(

**Comment by b-schweiger** [comment]
We would appreciate this feature. We're handling our waitlist manually anyway, so the missing automatic wouldn't be an issue for us. But I can see the automatic being important for others. 
One simple possibility would be to limit the number of tickets you can "order" per wait list entry. If you limit that to n, at most (n-1) tickets will not get sold. Since n=2 would be a sensible choice (larger groups should probably not put their faith in a wait list anyway), at most one ticket would not get sold. 
To make this a little more sophisticated, but still quite simple: Introduce a cutoff date (e.g. x days before the event starts), where tickets will be given out to people further down on the wait list if there are not enough tickets to fulfil the entries further up.
The next step could be to introduce a function that calculates the size of a waitlist entry that can be disregarded depending on the time to the event. For example: wait list entries for more than [DaysToEvent]/y may be disregarded. So at 100 days to the event, a single ticket will be given to the second entry on the wait list, if there is an entry for 10 tickets on top. 
If you want to make it even more complicated, you could introduce an AI predicting the number of returned tickets and the probabilty to fulfil a waitlist entry. But I guess, that would be overengineering...

**Comment by rabesocke** [comment]
We would love this enhancement / change of the waiting list. At the moment, the waiting list is not usable for us because most of the time at leat a couple or a group wants to participate at our events.

I would propose, that there will be three different options for the organiser:
1. First come, first serve - the first on the waiting list will get the notification when enough tickets are available
2. Best wins - the first best match on the waiting list will get the notification
3. Automation off - the organiser collect all the wishes on the waiting list and send the notification manually

**Comment by raphaelm** [comment]
Yeah, I think that's mostly the approach I'm leaning towards as well. There are some additional challenges (what if I want to wait on an adult + 1 child ticket? what if I want to wait for just any ticket but don't care which category?) that are *very* hard to put into simple UI, but ignoring these for now and adding just a "this many tickets" field, your approach would work.

Implementing the assignment algorithm efficiently will take some effort though.

**Comment by b-schweiger** [comment]
One possibility to make this easier to implement would be to enable the "multiple tickets" feature only with manual assignment.
